Documentation/quickstart/quickstart.rst

```diff
@@ -0,0 +1,110 @@
+
+   With NuttX, you compile the operating system and your application on your desktop or laptop computer, then install the
+   binary file on your embedded computer. This guide assumes your computer is an
+   `ARM <https://en.wikipedia.org/wiki/ARM_architecture>`_ CPU. If it isn't, you'll need a different tool chain.
+
+   You can download a toolchain from
+   `ARM Embedded Gnu Toolchain <https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm>`_
+   for your embedded processor's CPU. You can also use a toolchain shipped with your OS for the `none-eabi` target, such as `gcc-arm-none-eabi` in Linux.
+
+
+In the following example, we download ``gcc-arm-none-eabi`` version 9.0 and unpack it into ``/opt/gcc`:
+
+   .. code-block:: console
+
+      $ sudo mkdir /opt/gcc
+      $ sudo chgrp -R users /opt/gcc
+      $ cd /opt/gcc
+      $ wget https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2019q4/gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2
+      $ tar xf gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2
+      
+Then, add the toolchain ``bin/`` directory to your path:
+
+.. code-block:: console
+
+      $ echo "export PATH=/opt/gcc/gcc-arm-none-eabi-9-2019-q4-major/bin:$PATH" >> ~/.bashrc
+
+If you are using any other shell, the procedure is similar by editing the corresponding rc file.
+
+#. Download Apache NuttX
+The next step is to download NuttX main repository along the application repository. The latter is technically optional in a very minimal configurations but should be included in normal configuration since it includes the NuttX shell.
+   .. code-block:: console
+
+      $ mkdir nuttx
+      $ cd nuttx
+      $ git clone https://github.com/apache/incubator-nuttx.git nuttx
+      $ git clone https://github.com/apache/incubator-nuttx-apps apps
+      $ git clone https://bitbucket.org/nuttx/tools.git tools
+
+#. Install the ``kconfig-frontends`` package
+
+   NuttX is configured using ``kconfig`` system via an interactive menu system (``menuconfig``). It also includes the ``kconfig-tweak`` utility that can be used to quickly change debug settings without going into the menu system.
+
+   .. code-block:: console
+
+      $ cd tools/kconfig-frontends
+      $ # on MacOS do the following:
+      $ patch < ../kconfig-macos.diff -p 1
+      $ ./configure --enable-mconf --disable-shared --enable-static --disable-gconf --disable-qconf --disable-nconf
+      $ # on Linux do the following:
+      $  ./configure --enable-mconf --disable-nconf --disable-gconf --disable-qconf
+      $ make
+      $ make install
+
+#. List Possible Apache NuttX Base Configurations
+
+   Find your hardware and a good starting application in the list of base configurations. In the application list,
+   ``nsh`` is the Apache NuttX Shell, an interactive commandline that's a good starting place if you're new.
+
+   .. code-block:: bash
+
+      $ cd nuttx
+      $ ./tools/configure.sh list | less
```

Comment:
`list` doesn't exist as an option, that's actually printing an error then dumping all the configurations.

Reply:
You are right, there does not seem to be a "list" option.

Reply:
@Ouss4 @v01d Yes, I know I'm "misusing" the script here, but this does work, since if you put in an unknown option, it lists all the boards. Ideally we'd add an option to list the boards, but since this works now, is it ok to leave in? I could put a note or a todo. I'm also willing to add a `list` option to the script in another PR.

Reply:
If you add a todo so that we remember to update it with the expected use after adding this feature to configure script I think it is ok to leave as is.

Reply:
```bash
diff --git a/tools/configure.sh b/tools/configure.sh
index bf6702d115..a16dadbcaa 100755
--- a/tools/configure.sh
+++ b/tools/configure.sh
@@ -58,6 +58,17 @@ Where:
 
 "
 
+function dumpcfgs
+{
+  configlist=`find ${TOPDIR}/boards -name defconfig`
+  for defconfig in ${configlist}; do
+    config=`dirname ${defconfig} | sed -e "s,${TOPDIR}/boards/,,g"`
+    boardname=`echo ${config} | cut -d'/' -f3`
+    configname=`echo ${config} | cut -d'/' -f5`
+    echo "  ${boardname}:${configname}"
+  done
+}
+
 # A list of optional files that may be installed
 
 OPTFILES="\
@@ -95,6 +106,10 @@ while [ ! -z "$1" ]; do
   -e )
     distclean=y
     ;;
+  list )
+    dumpcfgs
+    exit 0
+    ;;
   -h )
     echo "$USAGE"
     exit 0
@@ -134,13 +149,7 @@ if [ ! -d ${configpath} ]; then
     echo "Directory for ${boardconfig} does not exist.  Options are:"
     echo ""
     echo "Select one of the following options for <board-name>:"
-    configlist=`find ${TOPDIR}/boards -name defconfig`
-    for defconfig in ${configlist}; do
-      config=`dirname ${defconfig} | sed -e "s,${TOPDIR}/boards/,,g"`
-      boardname=`echo ${config} | cut -d'/' -f3`
-      configname=`echo ${config} | cut -d'/' -f5`
-      echo "  ${boardname}:${configname}"
-    done
+    dumpcfgs
     echo ""
     echo "$USAGE"
     exit 3
```
We can do this.  `configure.c` needs to be updated too, however. (I guess we can do that separately, there are other places where it's lagging behind, like custom boards support).

Reply:
BTW, I agree on making the change on a separate PR.  Just one suggestion make it "--list" instead of just "list" as it is easier to adapt configure.c later.

Reply:
I created #1697 for that.  Please take a look.

Reply:
`-L` for listing is now merged.  Thanks @Ouss4

Reply:
Thank you @btashton @Ouss4 – I'll update the docs to use `-L` when I update this PR, should be able to do it tomorrow.

Reply:
@Ouss4 @btashton @v01d I updated the docs to use the new ``configure.sh -L`` option. 🙂 Thank you!